compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5899,26 +5899,39 @@ OMR::Power::TreeEvaluator::generateHelperBranchAndLinkInstruction(
       conditions, helperSym);
    }
 
+inline void generateLoadJ9Class(TR::Node* node, TR::Register* j9classReg, TR::Register *objReg, TR::CodeGenerator* cg)
+   {
+   if (TR::Compiler->om.compressObjectReferences())
+      generateTrg1MemInstruction(cg, TR::InstOpCode::lwz, node, j9classReg,
+         TR::MemoryReference::createWithDisplacement(cg, objReg, static_cast<int32_t>(TR::Compiler->om.offsetOfObjectVftField()), 4));
+   else
+      generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, j9classReg,
+         TR::MemoryReference::createWithDisplacement(cg, objReg, static_cast<int32_t>(TR::Compiler->om.offsetOfObjectVftField()), TR::Compiler->om.sizeofReferenceAddress()));
+   TR::TreeEvaluator::generateVFTMaskInstruction(cg, node, j9classReg);
+   }
+
 TR::Register *OMR::Power::TreeEvaluator::setmemoryEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    TR::Compilation *comp = cg->comp();
-   TR::Node             *dstAddrNode, *lengthNode, *valueNode;
-   dstAddrNode = node->getChild(0);
-   lengthNode = node->getChild(1);
-   valueNode = node->getChild(2);
 
-   TR::Register         *dstAddrReg, *lengthReg, *valueReg;
-   bool stopUsingCopyReg1, stopUsingCopyReg2 = false, stopUsingCopyReg3 = false;
+   TR::Node *dstBaseAddrNode = node->getChild(0);
+   TR::Node *dstOffsetNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   TR::Node *valueNode = node->getChild(3);
 
-   stopUsingCopyReg1 = TR::TreeEvaluator::stopUsingCopyReg(dstAddrNode, dstAddrReg, cg);
+   TR::Register         *dstBaseAddrReg, *dstOffsetReg, *lengthReg, *valueReg;
+   bool stopUsingCopyReg1, stopUsingCopyReg2, stopUsingCopyReg3 = false, stopUsingCopyReg4 = false;
+
+   stopUsingCopyReg1 = TR::TreeEvaluator::stopUsingCopyReg(dstBaseAddrNode, dstBaseAddrReg, cg);
+   stopUsingCopyReg2 = TR::TreeEvaluator::stopUsingCopyReg(dstOffsetNode, dstOffsetReg, cg);
```

Comment:
keep performance in mind when doing codegen works.  here, you seemed assuming an offset register is always needed, but it isn't true.  how about it being a constant fit-in 16bits?